src/ensemble/basic_ensemble_solve.jl

```diff
@@ -137,7 +137,16 @@ end
 
 function batch_func(i,prob,alg;kwargs...)
   iter = 1
-  _prob = prob.safetycopy ? deepcopy(prob.prob) : prob.prob
+
+  # if no user specified safetycopy and one thread we don't copy, otherwise
+  # we copy when given a user specified prob_func
+  if prob.safetycopy === nothing
+    safetycopy = (Threads.nthreads() == 1) ? false : DEFAULT_SAFETYCOPY(prob.prob_func)
```

Comment:
Maybe it would be cleaner to use
```suggestion
_safetycopy = if safetycopy === nothing
    Threads.nthreads() == 1 ? false : DEFAULT_SAFETYCOPY(prob.prob_func)
else
    safetycopy
end
```
in the keyword argument constructors of EnsembleProblem and use `safetycopy = nothing` as default keyword argument (as done in this PR)? Then one would not have to change the type of `safetycopy::Bool` to `safetycopy::Union{Nothing,Bool}`.

Reply:
Yeah, though I wonder if I should be checking if it would be better to just check if the user is using EnsembleSerial? Even if the number of threads is two, if they are using the serial algorithm we'd still want this fix right? In that case we can't make the decision at the time of constructing the JumpProblem based on `nthreads` since the choice of what algorithm to use occurs during the call to solve.

Reply:
Interesting. Is there a case where that is safer? Since you cannot change numbers of threads while Julia is running, I'm not sure if it makes a difference?

Reply:
EnsembleSerial already won't safetycopy IIRC.

Reply:
No it does if `prob_func` is not the default. That was @TorkelE problem.

Reply:
`safetycopy` is (currently) set before the choice of EnsembleAlg type.

Reply:
Shouldn't this be fixed in the implementation of EnsembleSerial (i.e., shouldn't it just ignore `safetycopy`)? It seems this is orthogonal to the default value of `safetycopy` of EnsembleProblem.

Reply:
Could there be contexts where a user might want to always force a copy even in serial, so we wouldn't want to override their default? If we want to say that it is only used with threading though that is fine and I can update to handle both cases.

Reply:
Yes. If a user has callbacks which mutates the parameters, you need to copy or the second solve will not match the firsr